click/core.py

```diff
@@ -700,6 +700,12 @@ def main(self, args=None, prog_name=None, complete_var=None,
             except (EOFError, KeyboardInterrupt):
                 echo(file=sys.stderr)
                 raise Abort()
+            except Exit as e:
+                if not standalone_mode:
+                    if e.exit_code:
+                        raise
```

Comment:
Why re-raise the exception? The exit code should just be the return value from `invoke`, whatever it means